homeassistant/components/mopar/__init__.py

```diff
@@ -1 +1,168 @@
-"""The mopar component."""
+"""Support for Mopar vehicles."""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.lock import DOMAIN as LOCK
+from homeassistant.components.sensor import DOMAIN as SENSOR
+from homeassistant.components.switch import DOMAIN as SWITCH
+from homeassistant.const import (
+    CONF_USERNAME,
+    CONF_PASSWORD,
+    CONF_PIN,
+    CONF_SCAN_INTERVAL
+)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.event import track_time_interval
+
+REQUIREMENTS = ['motorparts==1.1.0']
+
+DOMAIN = 'mopar'
+DATA_UPDATED = '{}_data_updated'.format(DOMAIN)
+
+_LOGGER = logging.getLogger(__name__)
+
+COOKIE_FILE = 'mopar_cookies.pickle'
+SUCCESS_RESPONSE = 'completed'
+
+SUPPORTED_PLATFORMS = [LOCK, SENSOR, SWITCH]
+
+DEFAULT_INTERVAL = timedelta(days=7)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_PIN): cv.positive_int,
+        vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_INTERVAL):
+            vol.All(cv.time_period, cv.positive_timedelta),
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+SERVICE_HORN = 'sound_horn'
+ATTR_VEHICLE_INDEX = 'vehicle_index'
+SERVICE_HORN_SCHEMA = vol.Schema({
+    vol.Required(ATTR_VEHICLE_INDEX): cv.positive_int
+})
+
+
+def setup(hass, config):
+    """Set up the Mopar component."""
+    import motorparts
+
+    cookie = hass.config.path(COOKIE_FILE)
+    try:
+        session = motorparts.get_session(
+            config[CONF_USERNAME],
+            config[CONF_PASSWORD],
+            config[CONF_PIN],
+            cookie_path=cookie
+        )
+    except motorparts.MoparError:
+        _LOGGER.error("Failed to login")
+        return False
+
+    data = hass.data[DOMAIN] = MoparData(hass, session)
+    data.update(now=None)
+
+    track_time_interval(
+        hass, data.update, config[CONF_SCAN_INTERVAL]
+    )
+
+    def handle_horn(call):
+        """Enable the horn on the Mopar vehicle."""
+        try:
+            motorparts.remote_command(
```

Comment:
We can replace this with `motorparts.horn` which calls `remote_command` under the hood. And that allows us to use `data.actuate` instead. And then we don't need to expose `data.session` anymore, so we can let `MoparData` handle all session interactions. :tada:

Reply:
Good call, I missed that in the last round of cleaning up.